strawberry/field.py

```diff
@@ -70,7 +70,7 @@ def field(
     description: Optional[str] = None,
     permission_classes: Optional[List[Type[BasePermission]]] = None,
     federation: Optional[FederationFieldParams] = None
-) -> Union[Callable, StrawberryField]:
```

Comment:
@BryceBeagle I wonder if we need this in future 🤔

Reply:
well hopefully it will just be `-> StrawberryField` in the future

Reply:
Also, in hindsight, `StrawberryField` _is_ callable, so this is a bit redundant

Reply:
what I _really_ should have done was
```python
def field(resolver: Optional[Callable[T]], ...) -> Union[Callable[T], StrawberryField]: ...
```

Reply:
I think that would still break mypy, if we don't update the plugin 😊

for reference `dataclasses.field` is not typed :)

Reply:
Yeah for sure. I think the long-term solution would be to update the mypy plugin _and_ annotate here. Then we can have our 🎂 and eat it too.